pkg/ccl/backupccl/restore_online.go

```diff
@@ -392,21 +411,24 @@ func (r *restoreResumer) waitForDownloadToComplete(
 			}
 			lastProgressUpdate = timeutil.Now()
 		}
+		// Signal the download job if it is waiting that we've polled and found work
+		// left for it to do.
+		select {
+		case ch <- struct{}{}:
```

Comment:
if the channel is length 1, which it is now, doesn't that guarantee that sendDownloadWorker will receive an empty struct and will always attempt to redownload the spans? If we make this chan length 0, then we'll hit the default case here until `sendDownloadWorker` is waiting on the chan. I think that would prevent sendDownloadWorker from always  redownloading the spans.

Reply:
heh, the chan is length zero. we're good.